vignettes/Treatment.Rmd

```diff
@@ -15,50 +15,188 @@ knitr::opts_chunk$set(
 ```
 
 ```{r setup}
-suppressPackageStartupMessages(library(ggplot2))
 library(malariasimulation)
 library(malariaEquilibrium)
 ```
 
-Set up some baseline parameters:
+# Introduction
+
+This vignette describes how to implement clinical drug treatments in malariasimulation.
+
+The malariasimulation package contains in-built parameters sets for three anti-malarial drugs:
+
+1.  `AL_params`: artemether-lumefantrine (AL)
+2.  `DHA_PQP_params`: dihydroartemisinin and piperaquine (DHA-PQP)
+3.  `SP_AQ_params`: sulfadoxine-pyrimethamine and amodiaquine (SP-AQ)
+
+While all these drugs can be used to treat malaria, DHA-PQP and SP-AQ remain in the body for some time following treatment, making them good candidates for chemoprevention in mass drug administrations (see [Mass Drug Administration](https://mrc-ide.github.io/malariasimulation/articles/MDA.html). Any of these drugs can be included in the parameter list using the `set_drugs()` function (see [drugs_parameters.R](https://github.com/mrc-ide/malariasimulation/blob/master/R/drug_parameters.R) for full parameter details).
+
+Each drug parameter set is a vector of length four, that represent the drug efficacy, the infectiousness following treatment relative to an untreated infection, and parameters that determine the protection against reinfection ($P$): shape ($w$) and scale ($\lambda$) using an exponential decrease as follows:
+
+$$P = e^{{(-t/\lambda)}^w}$$
+
+The $P$ for each drug following treatment through time is plotted below.
+
+```{r, echo = FALSE, fig.align = 'center', fig.height = 5, fig.width = 7}
+cols  <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
+
+calc_P <- function(w, lambda, t){
+  P <- exp(-(t/lambda)^w)
+}
+
+t <- 1:70
+
+P_matrix <- lapply(list(AL_params, DHA_PQP_params, SP_AQ_params), function(x){
+  calc_P(w = x[3], lambda = x[4], t = t)
+})
+
+plot(x = t, y = P_matrix[[1]], type = "l", col = cols[1],
+     xlab = "Days", ylab = expression(paste("Protection against reinfection (",italic(P),")")))
+grid()
+
+# 
+invisible(sapply(2:3, function(x){
+  points(x = t, y = P_matrix[[x]], type = "l", col = cols[x])
+}))
+
+# Add legend
+legend("topright", legend = c("AL","DHA-PQP","SP-AQ"), col = cols, lty = 1)
+
+```
+
+For more details, please see:
+
+Okell, L., Cairns, M., Griffin, J. *et al.* Contrasting benefits of different artemisinin combination therapies as first-line malaria treatments using model-based cost-effectiveness analysis. *Nat Commun* **5**, 5606 (2014). <https://doi.org/10.1038/ncomms6606>.
+
+## Functions
+
+Drug parameters can be incorporated into a complete parameter set:
 
 ```{r}
+Intro_parms <- set_drugs(parameters = get_parameters(), drugs = AL_params)
```

Comment:
I think to be super clear and align with other vignettes can we call the `get_parameters()` part of this on a seperate line first?